Assets/MixedRealityToolkit/MSBuild/Scripts/UnityProjectInfo.cs

```diff
@@ -0,0 +1,214 @@
+                    {
+                        Debug.LogError($"Failed to retrieve AsmDef for script assembly: {pair.Key}");
+                    }
+
+                    toAdd = new CSProjectInfo(availablePlatforms, Guid.NewGuid(), null, pair.Value, Application.dataPath.Replace("Assets", "MSBuild"));
+                }
+                else
+                {
+                    string guid = AssetDatabase.AssetPathToGUID(asmDefPath);
+                    if (!Guid.TryParse(guid, out Guid guidResult))
+                    {
+                        Debug.LogError($"Failed to get GUID of the AsmDef at '{asmDefPath}' for assembly: {pair.Key}");
+                    }
+                    else
+                    {
+                        guidResult = Guid.NewGuid();
+                    }
+
+                    AssemblyDefinitionAsset assemblyDefinitionAsset = AssetDatabase.LoadAssetAtPath<AssemblyDefinitionAsset>(asmDefPath);
+                    AssemblyDefinitionInfo assemblyDefinitionInfo = assemblyDefinitionAsset == null ? null : JsonUtility.FromJson<AssemblyDefinitionInfo>(assemblyDefinitionAsset.text);
+                    assemblyDefinitionInfo?.Validate(availablePlatforms);
+                    toAdd = new CSProjectInfo(availablePlatforms, guidResult, assemblyDefinitionInfo, pair.Value, Application.dataPath.Replace("Assets", "MSBuild"));
+                }
+
+                csProjects.Add(pair.Key, toAdd);
+            }
+
+            Plugins = new ReadOnlyDictionary<string, PluginAssemblyInfo>(ScanForPluginDLLs());
+
+            foreach (PluginAssemblyInfo plugin in Plugins.Values)
+            {
+                if (plugin.Type == PluginType.Native)
+                {
+                    Debug.LogWarning($"Native plugin {plugin.ReferencePath.AbsolutePath} not yet supported for MSBuild project.");
+                }
+            }
+
+            foreach (CSProjectInfo project in CSProjects.Values)
+            {
+                // Get the assembly references first from AssemblyDefinitionInfo if available (it's actually more correct), otherwise fallback to Assemby
+                IEnumerable<string> references = project.AssemblyDefinitionInfo == null
+                    ? project.Assembly.assemblyReferences.Select(t => t.name)
+                    : (project.AssemblyDefinitionInfo.references ?? Array.Empty<string>());
+
+                foreach (string reference in references)
+                {
+                    if (CSProjects.TryGetValue(reference, out CSProjectInfo dependency))
+                    {
+                        project.AddDependency(dependency);
+                    }
+                    else
+                    {
+                        Debug.LogError($"Failed to get dependency '{reference}' for project '{project.Name}'.");
+                    }
+                }
+
+
+                // TODO Find a better way to filter which plugins should be included; project.Assembly.allReferences isn't correct 
```

Comment:
What are the other ramifications of this TODO? It feels kinda scary to read this (i.e. this isn't really correct) - is it more like, it's not exactly optimal?

Reply:
Yeah, the auto-referenced.

---
In reply to: [302783219](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5215#discussion_r302783219) [](ancestors = 302783219)